src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -57,6 +59,44 @@ internal DateTime StartTimeCore
             }
         }
 
+        /// <summary>Computes a time based on a number of ticks since boot.</summary>
+        /// <param name="timespanAfterBoot">The timespan since boot.</param>
+        /// <returns>The converted time.</returns>
+        internal static DateTime BootTimeToDateTime(TimeSpan timespanAfterBoot)
+        {
+            // Use the uptime and the current time to determine the absolute boot time.
+            DateTime bootTime = DateTime.UtcNow - Uptime;
+
+            // And use that to determine the absolute time for timespan.
+            DateTime dt = bootTime + timespanAfterBoot;
+
+            // The return value is expected to be in the local time zone.
+            // It is converted here (rather than starting with DateTime.Now) to avoid DST issues.
+            return dt.ToLocalTime();
+        }
+
+        /// <summary>Gets the elapsed time since the system was booted.</summary>
+        private static TimeSpan Uptime
+        {
+            get
+            {
+                // '/proc/uptime' accounts time a device spends in sleep mode.
+                const string UptimeFile = Interop.procfs.ProcUptimeFilePath;
+                string text = File.ReadAllText(UptimeFile);
+
+                double uptimeSeconds = 0;
+                int length = text.IndexOf(' ');
+                Debug.Assert(length != -1);
```

Comment:
We don't seem to use Debug.Assert a lot in this codebase. Generally we either ignore a failure or we don't, we don't ignore in retail and assert in debug. If we see these asserts fire, what would you expect us to do? 

Note, we are probably going to focus our official builds on running tests on release only.

Reply:
This is expressing conditions that are expected to never fail. I can leave them out if you want.

Reply:
> I can leave them out if you want.

Yeah, let's remove them.  If it does fail, it's not an indication that there's something wrong in this code, and there's nothing we can do about it (we already have the code in place to handle that).